File: data-jpa/src/test/java/top/bettercode/summer/data/jpa/support/QuerydslJpaExtPredicateExecutorTest.java
package top.bettercode.summer.data.jpa.support;

import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.Optional;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Sort;
import org.springframework.test.context.junit.jupiter.SpringExtension;
import top.bettercode.summer.data.jpa.domain.QUser;
import top.bettercode.summer.data.jpa.domain.User;
import top.bettercode.summer.data.jpa.repository.UserRepository;

/**
 * @author Peter Wu
 */
@Disabled
@ExtendWith(SpringExtension.class)
@SpringBootTest
public class QuerydslJpaExtPredicateExecutorTest {

  @Autowired
  UserRepository repository;
  Integer daveId;
  Integer carterId;

  @BeforeEach
  public void setUp() {
    User dave = new User("Dave", "Matthews");
    repository.save(dave);
    User dave1 = new User("Dave", "Matthews");
    repository.save(dave1);
    User carter = new User("Carter", "Beauford");
    repository.save(carter);
    carter = new User("Carter", "Beauford");
    repository.save(carter);

    daveId = dave1.getId();

    repository.delete(dave);
    carterId = carter.getId();

  }

  @AfterEach
  public void tearDown() {
    repository.deleteAll();
    repository.cleanRecycleBin();
  }

  @Test
  public void findOne() {
    Optional<User> dave = repository.findOne(QUser.user.firstName.eq("Dave"));
    dave.ifPresent(System.out::println);
    assertTrue(dave.isPresent());
  }

  @Test
  public void findAll() {
    System.err.println(repository.findAll());
    Iterable<User> carter = repository.findAll(QUser.user.firstName.eq("Carter"));
    org.junit.jupiter.api.Assertions.assertTrue(carter.iterator().hasNext());
  }

  @Test
  public void findAll1() {
    org.junit.jupiter.api.Assertions.assertTrue(
        repository.findAll(QUser.user.firstName.eq("Carter"), QUser.user.lastName.asc()).iterator()
            .hasNext());
  }

  @Test
  public void findAll2() {
    org.junit.jupiter.api.Assertions.assertTrue(
        repository.findAll(QUser.user.firstName.eq("Carter"), Sort.by("id")).iterator().hasNext());
  }

  @Test
  public void findAll3() {
    org.junit.jupiter.api.Assertions.assertTrue(
        repository.findAll(QUser.user.lastName.asc()).iterator().hasNext());
  }

  @Test
  public void findAll4() {
    org.junit.jupiter.api.Assertions.assertEquals(1,
        repository.findAll(QUser.user.firstName.eq("Carter"), PageRequest.of(0, 1)).getContent()
            .size());
    org.junit.jupiter.api.Assertions.assertEquals(2,
        repository.findAll(QUser.user.firstName.eq("Carter"), PageRequest.of(0, 5)).getContent()
            .size());
  }

  @Test
  public void count() {
    org.junit.jupiter.api.Assertions.assertEquals(2,
        repository.count(QUser.user.firstName.eq("Carter")));
  }

  @Test
  public void exists() {
    org.junit.jupiter.api.Assertions.assertTrue(
        repository.exists(QUser.user.firstName.eq("Carter")));
  }

  @Test
  public void findOneFromRecycleBin() {
    Optional<User> dave = repository.findOneFromRecycleBin(QUser.user.firstName.eq("Dave"));
    dave.ifPresent(System.out::println);
    org.junit.jupiter.api.Assertions.assertTrue(dave.isPresent());
  }

  @Test
  public void findAllFromRecycleBin() {
    org.junit.jupiter.api.Assertions.assertTrue(
        repository.findAllFromRecycleBin(QUser.user.firstName.eq("Dave")).iterator().hasNext());
  }

  @Test
  public void findAllFromRecycleBin1() {
    org.junit.jupiter.api.Assertions.assertTrue(
        repository.findAllFromRecycleBin(QUser.user.firstName.eq("Dave"), QUser.user.lastName.asc())
            .iterator()
            .hasNext());
  }

  @Test
  public void findAllFromRecycleBin2() {
    org.junit.jupiter.api.Assertions.assertTrue(
        repository.findAllFromRecycleBin(QUser.user.firstName.eq("Dave"), Sort.by("id")).iterator()
            .hasNext());
  }

  @Test
  public void findAllFromRecycleBin3() {
    org.junit.jupiter.api.Assertions.assertTrue(
        repository.findAllFromRecycleBin(QUser.user.lastName.asc()).iterator().hasNext());
  }

  @Test
  public void findAllFromRecycleBin4() {
    repository.deleteById(daveId);
    org.junit.jupiter.api.Assertions.assertEquals(1,
        repository.findAllFromRecycleBin(QUser.user.firstName.eq("Dave"), PageRequest.of(0, 1))
            .getContent()
            .size());
    org.junit.jupiter.api.Assertions.assertEquals(2,
        repository.findAllFromRecycleBin(QUser.user.firstName.eq("Dave"), PageRequest.of(0, 5))
            .getContent()
            .size());
  }

  @Test
  public void countRecycleBin() {
    org.junit.jupiter.api.Assertions.assertEquals(1,
        repository.countRecycleBin(QUser.user.firstName.eq("Dave")));
  }

  @Test
  public void existsInRecycleBin() {
    org.junit.jupiter.api.Assertions.assertTrue(
        repository.existsInRecycleBin(QUser.user.firstName.eq("Dave")));
    org.junit.jupiter.api.Assertions.assertFalse(
        repository.existsInRecycleBin(QUser.user.firstName.eq("Carter")));
  }
}

File: data-jpa/src/test/kotlin/top/bettercode/summer/data/jpa/support/QuerydslJpaExtPredicateExecutorTest.kt
package top.bettercode.summer.data.jpa.support

import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Sort
import org.springframework.test.context.junit.jupiter.SpringExtension
import top.bettercode.summer.data.jpa.domain.QUser
import top.bettercode.summer.data.jpa.domain.User
import top.bettercode.summer.data.jpa.repository.UserRepository
import java.util.*

/**
 * @author Peter Wu
 */
//@Disabled
@ExtendWith(SpringExtension::class)
@SpringBootTest
class QuerydslJpaExtPredicateExecutorTest {
    @Autowired
    lateinit var repository: UserRepository
    var daveId: Int = 0
    var carterId: Int = 0

    @BeforeEach
    fun setUp() {
        val dave = User("Dave", "Matthews")
        repository.save(dave)
        val dave1 = User("Dave", "Matthews")
        repository.save(dave1)
        var carter = User("Carter", "Beauford")
        repository.save(carter)
        carter = User("Carter", "Beauford")
        repository.save(carter)
        daveId = dave1.id!!
        repository.delete(dave)
        carterId = carter.id!!
    }

    @AfterEach
    fun tearDown() {
        repository.deleteAll()
        repository.cleanRecycleBin()
    }

    @Test
    fun name() {
        System.err.println(repository.findAll(QUser.user.firstName!!.contains("D")))
    }

    @Test
    fun findOne() {
        val dave: Optional<User?> = repository.findOne(QUser.user.firstName.eq("Dave"))
        dave.ifPresent { x: User -> println(x) }
        Assertions.assertTrue(dave.isPresent)
    }

    @Test
    fun findAll() {
        System.err.println(repository.findAll())
        val carter: Iterable<User?> = repository.findAll(QUser.user.firstName!!.eq("Carter"))
        Assertions.assertTrue(carter.iterator().hasNext())
    }

    @Test
    fun findAll1() {
        Assertions.assertTrue(
                repository.findAll(QUser.user.firstName!!.eq("Carter"), QUser.user.lastName!!.asc()).iterator()
                        .hasNext())
    }

    @Test
    fun findAll2() {
        Assertions.assertTrue(
                repository.findAll(QUser.user.firstName!!.eq("Carter"), Sort.by("id")).iterator().hasNext())
    }

    @Test
    fun findAll3() {
        Assertions.assertTrue(
                repository.findAll(QUser.user.lastName!!.asc()).iterator().hasNext())
    }

    @Test
    fun findAll4() {
        Assertions.assertEquals(1,
                repository.findAll(QUser.user.firstName!!.eq("Carter"), PageRequest.of(0, 1)).content
                        .size)
        Assertions.assertEquals(2,
                repository.findAll(QUser.user.firstName.eq("Carter"), PageRequest.of(0, 5)).content
                        .size)
    }

    @Test
    fun count() {
        Assertions.assertEquals(2,
                repository.count(QUser.user.firstName!!.eq("Carter")))
    }

    @Test
    fun exists() {
        Assertions.assertTrue(
                repository.exists(QUser.user.firstName!!.eq("Carter")))
    }

    @Test
    fun findOneFromRecycleBin() {
        val dave: Optional<User> = repository.findOneFromRecycleBin(QUser.user.firstName!!.eq("Dave"))
        dave.ifPresent { x: User? -> println(x) }
        Assertions.assertTrue(dave.isPresent)
    }

    @Test
    fun findAllFromRecycleBin() {
        Assertions.assertTrue(
                repository.findAllFromRecycleBin(QUser.user.firstName!!.eq("Dave")).iterator().hasNext())
    }

    @Test
    fun findAllFromRecycleBin1() {
        Assertions.assertTrue(
                repository.findAllFromRecycleBin(QUser.user.firstName!!.eq("Dave"), QUser.user.lastName!!.asc())
                        .iterator()
                        .hasNext())
    }

    @Test
    fun findAllFromRecycleBin2() {
        Assertions.assertTrue(
                repository.findAllFromRecycleBin(QUser.user.firstName!!.eq("Dave"), Sort.by("id")).iterator()
                        .hasNext())
    }

    @Test
    fun findAllFromRecycleBin3() {
        Assertions.assertTrue(
                repository.findAllFromRecycleBin(QUser.user.lastName!!.asc()).iterator().hasNext())
    }

    @Test
    fun findAllFromRecycleBin4() {
        repository.deleteById(daveId)
        Assertions.assertEquals(1,
                repository.findAllFromRecycleBin(QUser.user.firstName!!.eq("Dave"), PageRequest.of(0, 1))
                        .content
                        .size)
        Assertions.assertEquals(2,
                repository.findAllFromRecycleBin(QUser.user.firstName.eq("Dave"), PageRequest.of(0, 5))
                        .content
                        .size)
    }

    @Test
    fun countRecycleBin() {
        Assertions.assertEquals(1,
                repository.countRecycleBin(QUser.user.firstName!!.eq("Dave")))
    }

    @Test
    fun existsInRecycleBin() {
        Assertions.assertTrue(
                repository.existsInRecycleBin(QUser.user.firstName!!.eq("Dave")))
        Assertions.assertFalse(
                repository.existsInRecycleBin(QUser.user.firstName.eq("Carter")))
    }
}